scripts/formater.pl

```diff
@@ -32,21 +32,29 @@
 #
 # NP: The Squid code requires astyle version 2.04 (exactly for now)
 #
-my $ASTYLE_BIN="/usr/local/bin/astyle";
-if (! -x $ASTYLE_BIN) {
-  $ASTYLE_BIN="/usr/bin/astyle";
-}
-if (! -x $ASTYLE_BIN) {
-  $ASTYLE_BIN="/usr/local/src/astyle-2.04/bin/astyle";
+my @ASTYLE_PATHS=(
+    $ENV{'ASTYLE'},
```

Comment:
This list should not exist, but if you keep it, please at least avoid adding an undefined value to it.

Even if you change conditional export in the shell script, please do not assume that this Perl script is always executed from that shell script. This Perl script is very handy as a stand-alone tool as well.

Reply:
and it will work as it does now, no worse. Not adding the undefined entry would require extra code complexity

Reply:
> Not adding the undefined entry would require extra code complexity

Actually, _adding_ this undefined value already caused "extra code complexity" and, naturally, a [bug](https://github.com/squid-cache/squid/pull/644/files#r432835819) (both in the new loop iterating this deceivingly "simple" array). However, this is still your call. I am simply pointing out problems. The code has many other problems. I am not blocking this unfortunate change of an already bad code.